### DISCUSSION TITLE
How to invalidate graph's data and redraw graph?

### DISCUSSION THREAD

**Discussion by vivekgupta4Git** [patrykandpatrick/vico#777] [Questions]
### Question

I am using single '**CartesinaChartHost**' with multiple parameters for which I need to draw graphs (see Screenshot)
<img width="370" alt="Screenshot 2024-06-28 at 8 56 55 AM" src="https://github.com/patrykandpatrick/vico/assets/91813403/11b5cfee-9854-4e95-8448-acc8d1e36700">
For some parameters there is data available and in the **viewModel** I am running the transaction to generate the series without any issues. However, for other parameters there might be no data, resulting in no series generation. 
`if (graphData.isNotEmpty()) {
                    modelProducer.tryRunTransaction {
                        lineSeries {
                            series(graphData.keys, graphData.values)
                        }
                    }
                }`
as there is no series generated graph remains the same for the new parameter but current workaround is to generate a single value.
`modelProducer.tryRunTransaction {
                        lineSeries {
                            series(x = listOf(0f), y = listOf(0))
                        }
                    }`
The problem with this approach is that sometimes it crashes with an error saying "Cannot round NaN," which occurs inside the **shapeComponent's** draw method.
Is there any way to invalidate the graph for each parameter, ensuring it updates correctly even when there is no data?

### Vico version(s)

2.0.0-alpha.21

### UI framework(s)

Jetpack Compose

**Comment by Gowsky** [comment]
Hello, @vivekgupta4Git.
>Is there any way to invalidate the graph for each parameter, ensuring it updates correctly even when there is no data?

Yes, please see #818. 

>The problem with this approach is that sometimes it crashes with an error saying "Cannot round NaN," which occurs inside the shapeComponent's draw method.

If this persists, please share more details on how to reproduce it.
